src/main/java/ru/hogwarts/school/service/AvatarService.java

```diff
@@ -0,0 +1,65 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import ru.hogwarts.school.model.Avatar;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repository.AvatarRepository;
+import ru.hogwarts.school.repository.StudentRepository;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+
+@Service
+public class AvatarService {
+    private final StudentService studentService;
+    private final AvatarRepository avatarRepository;
+    @Value("${students.avatars.dir.path}")
+    private String avatarsDir;
+
+    public AvatarService(StudentService studentService, AvatarRepository avatarRepository, StudentRepository studentRepository) {
+        this.studentService = studentService;
+        this.avatarRepository = avatarRepository;
+    }
+
+    public void uploadAvatar(Long studentId, MultipartFile avatarFile) throws IOException {
+
+        Student student = studentService.findStudent(studentId);
+        Path filePath = Path.of(avatarsDir, student + "." + getExtensions(avatarFile.getOriginalFilename()));
+        Files.createDirectories(filePath.getParent());
+        Files.deleteIfExists(filePath);
+        try (
+                InputStream is = avatarFile.getInputStream();
+                OutputStream os = Files.newOutputStream(filePath, CREATE_NEW);
+                BufferedInputStream bis = new BufferedInputStream(is, 1024);
+                BufferedOutputStream bos = new BufferedOutputStream(os, 1024);
+        ) {
+            bis.transferTo(bos);
+        }
+
+        Avatar avatar = findAvatar(studentId);
+        avatar.setStudent(student);
+        avatar.setFilePath(filePath.toString());
+        avatar.setFileSize(avatarFile.getSize());
+        avatar.setMediaType(avatarFile.getContentType());
+        avatar.setData(avatarFile.getBytes());
+        avatarRepository.save(avatar);
+    }
+
+    private String getExtensions(String fileName) {
+        return fileName.substring(fileName.lastIndexOf(".") + 1);
+    }
+
+    public Avatar findAvatar(Long studentId) {
+        return avatarRepository.findByStudentId(studentId).orElseThrow();
+    }
+
+    public Collection<Avatar> getAllAvatars(Integer pageNumber, Integer pageSize) {
+        return avatarRepository.getAll(pageNumber, pageSize);
```

Comment:
Pageable pageable = PageRequest.of(page, size);
Page<Avatar> avatars = avatarRepository.findAll(pageable);
return avatars.stream().toList();